pep-0690.rst

```diff
@@ -312,57 +316,73 @@ import object. When ``modb.foo`` is later referenced, it will also try to
 class or function bodies is handled in the compiler via a new
 ``EAGER_IMPORT_NAME`` opcode that always imports eagerly. Top-level imports use
 ``IMPORT_NAME``, which may be lazy or eager depending on ``-L`` and/or
 ``importlib.set_lazy_imports()``.
 
+The current status of lazy imports at any given place is determined by a
+combination of the passed ``-L`` option flag; an interpreter-wide flag set by
+``importlib.set_lazy_imports()`` and the container object passed in its
+``excluding`` keyword argument; and a flag set by
+``importlib.enable_lazy_imports_in_module()`` in nearest running module frame.
+All these together are used to cache the current status of lazy imports in the
+currently running frame, this cache is globally busted whenever any of these API
+functions is called and, as such, changes are immediate for all new import
+statements.
 
-Exceptions
-----------
+
+Exceptions and Deferred Exceptions
+----------------------------------
 
 Exceptions that occur during a lazy import bubble up and erase the
 partially-constructed module(s) from ``sys.modules``, just as exceptions during
 normal import do.
 
 Since errors raised during a lazy import will occur later (wherever the imported
 name is first referenced) than they would if the import were eager, it is
-possible that they could be accidentally caught by exception handlers that
+also possible that they could be accidentally caught by exception handlers that
 didn't expect the import to be running within their ``try`` block, leading to
-confusion. To reduce the potential for this confusion, exceptions raised in the
-course of executing a lazy import will be replaced by a ``LazyImportError``
-exception (a subclass of ``ImportError``), with ``__cause__`` set to the
-original exception.
-
-The ``LazyImportError`` will have source location metadata attached pointing the
-user to the original import statement, to ease debuggability of errors from lazy
-imports. (It won't have a full traceback to the original import location; this
-is too expensive to preserve for all lazy imports, and it's not clear that it
-provides significant value over simply knowing the location of the import
-statement.)
+confusion.
 
-Only ``Exception`` are replaced in this way, not ``BaseException``.
-``BaseException`` are for "system-exiting" exceptions like ``KeyboardInterrupt``
-or ``SystemExit``; these are normally not caught, and if they are caught, it is
-less likely to be specific to a certain bit of code that was expected to raise
-them, and more likely that the goal is to catch them whatever their origin.
+Care should be taken when debugging unexpected exceptions keeping in mind any
+and all exceptions arising from import (including ``ModuleNotFoundError``) are
+deferred from import time to first-use time, which could complicate debugging.
+Referencing a name in the middle of any code could trigger a deferred import and
+produce an exception while loading and executing the related imported module.
```

Comment:
I don't think this paragraph adds any value over what is already said in the previous paragraph; I would just remove it. "Care should be taken" is not useful advice, and the rest of the paragraph is basically just repeating what the previous paragraph said.